node/bft/src/primary.rs

```diff
@@ -96,6 +98,8 @@ pub struct Primary<N: Network> {
     handles: Arc<Mutex<Vec<JoinHandle<()>>>>,
     /// The lock for propose_batch.
     propose_lock: Arc<TMutex<u64>>,
+    /// The outstanding requests to a peer.
+    outstanding_certificate_requests: Arc<RwLock<HashMap<(SocketAddr, u64), u32>>>,
```

Comment:
```suggestion
    outstanding_certificate_requests: Arc<Mutex<HashMap<(SocketAddr, u64), u32>>>,
```
since we never take advantage of concurrent `read`s